### PR TITLE
ARM: Cortex-M: add Cortex-M23 board target

### PR DESCRIPTION
**Pull request by ioannisg**
Completes build coverage for Cortex-M variants :)

Follow up: add TF-M integration for Baseline Cortex-M, since this is already a supported emulator in TF-M.
Possibly requires assembly adaptations in cortex-m code.

### CODE REVIEW

File: boards/arm/mps2_an519/Kconfig.board [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2021 Linaro Limited
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_MPS2_AN519
+	bool "Arm Cortex-M23 (Corstone-300, SSE-300 w/Ethos-U55) on MPS3 (AN547)"
```
Comment (galak) [inline boards/arm/mps2_an519/Kconfig.board:5]: need to fixup the comment string here

File: boards/arm/mps2_an519/mps2_an519_defconfig [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# Serial
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
+CONFIG_SERIAL=y
+CONFIG_UART_CMSDK_APB=y
+
+# Build a Secure firmware image
+CONFIG_TRUSTED_EXECUTION_SECURE=y
+
+CONFIG_I2C=y
```
Comment (gmarull) [inline boards/arm/mps2_an519/mps2_an519_defconfig:31]: is i2c needed by default config?

File: boards/arm/mps2_an519/mps2_an519-common.dtsi [hunk 3]
```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) 2019 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+ sysclk: system-clock {
```
Comment (gmarull) [inline boards/arm/mps2_an519/mps2_an519-common.dtsi:7]:
```suggestion
sysclk: system-clock {
```

File: boards/arm/mps2_an519/doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+     :width: 666px
+     :align: center
+     :height: 546px
+     :alt: ARM MPS3 AN547
+
+This board configuration also supports using the `Corstone-300 FVP`_ to emulate
+a MPS3 AN547 hardware platform.
+
+The Corstone-300 FVP (Fixed Virtual Platform) is a complete simulation of the
+Arm system, including processor, memory and peripherals. It is a available free
```
Comment (gmarull) [inline boards/arm/mps2_an519/doc/index.rst:29]:
```suggestion
Arm system, including processor, memory and peripherals. It is available free
```

File: boards/arm/mps2_an519/Kconfig.board (hunk 1 above)
Comment (ioannisg) [reply boards/arm/mps2_an519/Kconfig.board]: done

File: boards/arm/mps2_an519/mps2_an519_defconfig (hunk 2 above)
Comment (ioannisg) [reply boards/arm/mps2_an519/mps2_an519_defconfig]: nope. removed

File: boards/arm/mps2_an519/doc/index.rst (hunk 4 above)
Comment (ioannisg) [reply boards/arm/mps2_an519/doc/index.rst]: This part was removed, so this is resolved.

File: boards/arm/mps2_an519/mps2_an519-common.dtsi (hunk 3 above)
Comment (ioannisg) [reply boards/arm/mps2_an519/mps2_an519-common.dtsi:7]: done, thanks.

**Review by gmarull — APPROVED** [review]
A couple of minor nitpicks, +1 for board and docs part

**Comment by ioannisg** [comment]
@gmarull will address your comments, thanks.
@galak @microbuilder could you take a brief look? 
@galak comments on the assembly fixes are welcome.

File: boards/arm/mps2_an519/doc/index.rst [hunk 5]
```diff
@@ -0,0 +1,374 @@
...
+Connect the MPS2+ to your host computer using the USB port. You should see a
+USB connection exposing a Mass Storage (``V2M_MPS2`` by default).
+
+The update requires 3 steps:
+
+1. Copy application files to ``<MPS2 device name>/SOFTWARE/``.
+2. Open ``<MPS2 device name>/MB/HBI0263C/AN519/images.txt``.
+3. Update the ``AN519/images.txt`` file as follows:
+
+.. code-block:: bash
```
[the remark this answers is not in the document]
Comment (ioannisg) [reply boards/arm/mps2_an519/doc/index.rst:344]: Fixed.
Will add a separate commit to update this (as this was a copy past from another mps2 .rst)

File: boards/arm/mps2_an519/doc/index.rst [hunk 6]
```diff
@@ -0,0 +1,374 @@
...
+   - SPI_4_MOSI : D37
+   - SPI_4_MISO : D38
+   - SPI_4_SCK : D39
+   - I2C_4_SDA : D40
+   - I2C_4_SCL : D41
+
+For mode details refer to `MPS2+ AN519 Technical Reference Manual (TRM)`_.
+
+LED
+============
```
[the remark this answers is not in the document]
Comment (ioannisg) [reply boards/arm/mps2_an519/doc/index.rst]: fixed, thanks.

**Comment by microbuilder** [comment]
I've done a quick build test, but it's unfortunate that QEMU doesn't support this board yet for a quick test run, and I didn't see any M23 boards in the list:

```bash
$ qemu-system-arm -machine help | grep mps2
mps2-an385           ARM MPS2 with AN385 FPGA image for Cortex-M3
mps2-an386           ARM MPS2 with AN386 FPGA image for Cortex-M4
mps2-an500           ARM MPS2 with AN500 FPGA image for Cortex-M7
mps2-an505           ARM MPS2 with AN505 FPGA image for Cortex-M33
mps2-an511           ARM MPS2 with AN511 DesignStart FPGA image for Cortex-M3
mps2-an521           ARM MPS2 with AN521 FPGA image for dual Cortex-M33
```

**Review by microbuilder — APPROVED** [review]
Confirmed with a build test. Limited means to test further with no QEMU support available, at present.

**Review by MaureenHelm — APPROVED** [review]

**Comment by cfriedt** [comment]
@ioannisg - the test that is currently failing in CI passes when rebased on main. Can you please rebase?

```
zephyr % twister -p mps2_an521 -s tests/arch/arm/arm_interrupt/arch.interrupt.no_optimizations  
Renaming output directory to /Users/cfriedt/workspace/zephyrproject/zephyr/twister-out.8
INFO    - Zephyr version: v2.7.0-rc3-217-gad916a8c906a
INFO    - JOBS: 8
INFO    - Using 'zephyr' toolchain.
INFO    - Building initial testcase list...
INFO    - 1 test scenarios (1 configurations) selected, 0 configurations discarded due to filters.
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue
INFO    - Total complete:    1/   1  100%  skipped:    0, failed:    0
INFO    - 1 of 1 test configurations passed (100.00%), 0 failed, 0 skipped with 0 warnings in 51.87 seconds
INFO    - In total 4 test cases were executed, 0 skipped on 1 out of total 398 platforms (0.25%)
INFO    - 1 test configurations executed on platforms, 0 test configurations were only built.
INFO    - Saving reports...
INFO    - Writing xunit report /Users/cfriedt/workspace/zephyrproject/zephyr/twister-out/twister.xml...
INFO    - Writing xunit report /Users/cfriedt/workspace/zephyrproject/zephyr/twister-out/twister_report.xml...
INFO    - Run completed
```

**Comment by cfriedt** [comment]
> @ioannisg - the test that is currently failing in CI passes when rebased on main. Can you please rebase?

I was able to do it - should pass CI now 🤞

**Pull request closed without merging**